TypeCobol/Compiler/CodeModel/SymbolTable.cs

```diff
@@ -999,46 +999,52 @@ public List<FunctionDeclaration> GetFunction(QualifiedName name, ParameterList p
-
-            for (int c = 0; c < p1.InputParameters.Count; c++)
-                if (!TypeCompare(p1.InputParameters[c], p2.InputParameters[c], pgmName)) return false;
-            for (int c = 0; c < p1.InoutParameters.Count; c++)
-                if (!TypeCompare(p1.InoutParameters[c], p2.InoutParameters[c], pgmName)) return false;
-            for (int c = 0; c < p1.OutputParameters.Count; c++)
-                if (!TypeCompare(p1.OutputParameters[c], p2.OutputParameters[c], pgmName)) return false;
+            if (expected.InputParameters.Count != actual.Inputs.Count) return false;
+            if (expected.InoutParameters.Count != actual.Inouts.Count) return false;
+            if (expected.OutputParameters.Count != actual.Outputs.Count) return false;
+
+            for (int c = 0; c < expected.InputParameters.Count; c++)
+                if (!TypeCompare(expected.InputParameters[c], actual.Inputs[c], pgmName)) return false;
+            for (int c = 0; c < expected.InoutParameters.Count; c++)
+                if (!TypeCompare(expected.InoutParameters[c], actual.Inouts[c], pgmName)) return false;
+            for (int c = 0; c < expected.OutputParameters.Count; c++)
+                if (!TypeCompare(expected.OutputParameters[c], actual.Outputs[c], pgmName)) return false;
             return true;
         }
-        /// <summary>
-        /// Type comparaison
-        /// </summary>
-        /// <param name="p1">Represent the DataType from the called function/procedure</param>
-        /// <param name="p2">Represent the DataType from the caller function/procedure</param>
-        /// <param name="pgmName">Corresponds to the progam containing the called function/procedure</param>
-        /// <returns></returns>
-        private bool TypeCompare(ParameterDescription p1, DataType p2, string pgmName)
+
+        private bool TypeCompare(ParameterDescription parameter, TypeInfo argumentTypeInfo, string pgmName)
         {
             //Omitted accepted only if parameter is Omittable
-            if (p2 == DataType.Omitted) {
-                return p1.IsOmittable;
+            if (argumentTypeInfo.DataType == DataType.Omitted)
+            {
+                return parameter.IsOmittable;
             }
-            
-            var p1Types = p1.Name.Contains(".") ? this.GetType(p1) : this.GetType(p1.DataType, pgmName);
-            var p2Types = this.GetType(p2);
 
-            if (p1Types.Count > 1 || p2Types.Count > 1)
-                return false; //Means that a type is declare many times. Case already handle by checker.
-            var p1Type = p1Types.FirstOrDefault();
-            var p2Type = p2Types.FirstOrDefault();
+            var parameterType = parameter.TypeDefinition;
+            if (parameterType == null)
+            {
+                var parameterTypes = parameter.Name.Contains(".") ? this.GetType(parameter) : this.GetType(parameter.DataType, pgmName);
```

Comment:
Is the call to `GetType` still necessary here?

Reply:
This is required for ParameterDescription coming from a dependency. At this step, the TypeDefinition may not be resolved yet.